frontend/src/pages/Home/UploadPanel.tsx

```diff
@@ -1,21 +1,115 @@
 import React from 'react';
+import { useState } from 'react';
 import { Button } from "../../components/ui/button";
+import { useNavigate } from 'react-router-dom';
+
+// what about other types of files like png, mp4
+enum FileType {
+  Html = 'Html',
+  Css = 'Css',
+  Js = 'Js'
+}
+enum Framework {
+  VanillaProject = 'VanillaProject', 
+  React = 'React',
+  Angular = 'Angular',
+  Vue = 'Vue'
+}
+
+interface Page {
+  readonly pageId: string;
+  filePath: string; // full/file/path/file.extension
+  viewport: {
+    width: number,
+    height: number
+  };
+  pageContent: {
+    fileType: FileType;
+    framework: string | Framework;
+    body: {
+      originalVersion: string; // original code
+      transpiledVersion: string; // code converted into vanilla version
+    }
+  };
+}
 
 
 export function UploadPanel() {
+  // Array of uploaded files
+  const [files, setFiles] = useState<File[]>([]);
+  const navigate = useNavigate();
+
+  // When user upload files, add it to the files array
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>):void => {
+    if (e.target.files) {
+      // TODO: change Array.from() to push
+      setFiles(Array.from(e.target.files));
+    }
+  };
+
+  // When user click the enhance button
+  // Convert all files into Pages
+  // Send Pages to scan and navigate to it
+  const handleSubmit = async (): Promise<void> => {
+    const pages = await convertFilesToPages();
+    //console.log(files[0].type);
+    console.log(pages);
+    navigate('/output', { state: { pages } });
+  };
+
+  // Convert all uploaded files into interface Pages 
+  const convertFilesToPages = async (): Promise<Page[]> => {
+    // Get pages from files
+    const pages = await Promise.all(files.map(async (file) => {
+      // TODO: add type type
+      const fileContent = await file.text();
```

Comment:
package.json files should be JSON.stringify(). Right now the output page shows that package.json are in base64. They should be in a stringfied form of the JSON instead